roles/db-copy/tasks/active-copy.yml

```diff
@@ -0,0 +1,155 @@
+  environment:
+    ORACLE_HOME: "{{ oracle_home }}"
+    ORACLE_SID: "{{ oracle_sid }}"
+    PATH: "{{ oracle_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  delegate_to: primary1
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Add static listener entry
+  lineinfile:
+    path: "{{ grid_home }}/network/admin/listener.ora"
+    regexp: "^SID_LIST_{{ listener_name }}"
+    line: "SID_LIST_{{ listener_name }}=(SID_LIST=(SID_DESC=(GLOBAL_DBNAME={{ oracle_sid }}{{ standby_suffix }}.{{ db_domain }})(ORACLE_HOME={{ oracle_home }})(SID_NAME={{ oracle_sid }})))"
+    owner: "{{ grid_user }}"
+    group: "{{ oracle_group }}"
+  become: yes
+  tags: active-duplicate
+
+- name: Active-copy | Reload listener
+  shell: "{{ grid_home }}/bin/lsnrctl reload {{ listener_name }}"
+  environment:
+    ORACLE_HOME: "{{ grid_home }}"
+    PATH: "{{ grid_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  become: yes
+  become_user: "{{ grid_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Create directories
+  file:
+    path: "{{ oracle_base }}/admin/{{ oracle_sid }}/adump"
+    state: directory
+    owner: "{{ oracle_user }}"
+    group: "{{ oracle_group }}"
+  become: yes
+  tags: active-duplicate
+
+- name: Active-copy | Create auxiliary init file
+  shell: |
+    set -o pipefail
+    (echo "db_name={{ db_name }}"; echo "db_unique_name={{ db_name }}{{ standby_suffix }}") > {{ oracle_home }}/dbs/init{{ oracle_sid }}.ora
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Start auxiliary instance
+  shell: |
+    set -o pipefail
+    {{ oracle_home }}/bin/sqlplus -s / as sysdba << EOF
+    startup nomount pfile={{ oracle_home }}/dbs/init{{ oracle_sid }}.ora force
+    alter system register;
+    host sleep 60
+    EOF
+  environment:
+    ORACLE_HOME: "{{ oracle_home }}"
+    ORACLE_SID: "{{ oracle_sid }}"
+    PATH: "{{ oracle_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Set sys password for auxiliary instance
+  shell: |
+    set -o pipefail
+    {{ oracle_home }}/bin/orapwd file={{ oracle_home }}/dbs/orapw{{ db_name }} force=y password={{ sys_pass }}
+  environment:
+    ORACLE_HOME: "{{ oracle_home }}"
+    ORACLE_SID: "{{ oracle_sid }}"
+    PATH: "{{ oracle_home }}/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+  become: yes
+  become_user: "{{ oracle_user }}"
+  tags: active-duplicate
+
+- name: Active-copy | Duplicate primary database
+  shell: |
+    set -o pipefail
+    {{ oracle_home }}/bin/rman << EOF
+    connect target "sys/{{ sys_pass }}@//{{ lookup('env', 'PRIMARY_IP_ADDR') }}:{{ listener_port | default(1521, true) }}/{{ db_name }}{% if db_domain|default('', true)|length > 0 %}.{{ db_domain }}{% endif %}"
```

Comment:
To double-check:  will this log the sys password in output?

Reply:
if connections are within contents spooled to rman then ps won't show it.

Reply:
pythiangoryunov mentioned offline:  there is a step later where output is logged, and I believe it might contain a password

FWIW, when I test with 19c, ps doesn't show any command-line arguments:

$ rman target a/a &
...
$ cat /proc/$(pgrep rman)/cmdline
rman

Reply:
still working on it. Using shell module either I put connect target inside of EOF or use rman target ... it still showed through ps.

Reply:
I changed to use command module and provided connection strings as parameters and was able to avoid parameters shown through ps.